src/components/cart.js

```diff
@@ -317,11 +317,15 @@ export default class Cart extends Component {
    * Remove all lineItems in the cart
    */
   empty() {
-    return this.model.clearLineItems().then(() => {
+    const lineItemIds = this.model.lineItems ? this.model.lineItems.map((item) => item.id) : [];
+
+    this.model = this.props.client.removeLineItems(this.model.id, lineItemIds).then(() => {
       this.view.render();
       this.toggles.forEach((toggle) => toggle.view.render());
       return;
     });
+
+    return this.model;
```

Comment:
so model is a promise now?

Reply:
my thought was more

```javascript
return this.props.client.removeLineItems(this.model.id, lineItemIds).then((updatedCheckout) => {
  this.view.render();
  this.toggles.forEach((toggle) => toggle.view.render());
  this.model = updatedCheckout;

  return this.model;
});
```

That way, anyone calling `empty` will have access to the next cart, and the component itself will update.